activerecord/lib/arel/visitors/to_sql.rb

```diff
@@ -930,7 +930,8 @@ def prepare_update_statement(o)
             stmt.limit = nil
             stmt.offset = nil
             stmt.orders = []
-            stmt.wheres = [Nodes::In.new(o.key, [build_subselect(o.key, o)])]
+            columns = o.key.is_a?(Array) ? Arel::Nodes::Grouping.new(o.key) : o.key
```

Comment:
This is a smell. We should build a new node when building the AST instead of pushing the logic to the visitor. Can you take a look if we can build a different node instead of having to check the type of the key here?

Reply:
Any updated on a different approach?

Reply:
Sorry, I lost track of this PR. 
I'll try to get back to it this week

Reply:
I changed the implementation but it's not drastically different from the original one. The main idea is to prevent the visitor from caring about whether the key is composite or not and simply always use `Grouping` node for the `SELECT (<column_or_columns>) IN ()` subselect query

Having additional parenthesis doesn't change the query and unlikely will result in any noticeable performance impact